File: 2025/ts/src/lib/renderer.ts
import { execSync } from "child_process";
import * as path from "path";

const RENDERER_DIR = "../render";

export function PopulateRenderer(dayPart: string, value: number): void {
  const rendererPath = path.resolve(RENDERER_DIR);

  const cmd = `go run main.go populate ts ${dayPart} ${value}`;

  try {
    execSync(cmd, {
      cwd: rendererPath,
      stdio: "inherit",
    });
  } catch (err) {
    console.error(`Error running populate command: ${err}`);
    throw err;
  }
}

File: 2025/go/lib/renderer.go
package lib

import (
	"fmt"
	"os"
	"os/exec"
	"path/filepath"
)

const (
	RENDERER_DIR = "../render"
)

func PopulateRenderer(dayPart string, value int) error {
	rendererPath, err := filepath.Abs(RENDERER_DIR)
	if err != nil {
		return fmt.Errorf("error resolving renderer path: %v", err)
	}

	cmd := exec.Command("go", "run", "main.go", "populate", "go", dayPart, fmt.Sprintf("%d", value))
	cmd.Dir = rendererPath
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr

	if err := cmd.Run(); err != nil {
		return fmt.Errorf("error running populate command: %v", err)
	}

	return nil
}
